pkg/services/object/acl/acl.go

```diff
@@ -221,41 +224,35 @@ func (c *Checker) CheckEACL(msg any, reqInfo v2.RequestInfo) error {
 // isValidBearer checks whether bearer token was correctly signed by authorized
 // entity. This method might be defined on whole ACL service because it will
 // require fetching current epoch to check lifetime.
-func isValidBearer(reqInfo v2.RequestInfo, st netmap.State) error {
-	ownerCnr := reqInfo.ContainerOwner()
-
-	token := reqInfo.Bearer()
-
-	// 0. Check if bearer token is present in reqInfo.
-	if token == nil {
-		return nil
-	}
-
+func isValidBearer(token bearer.Token, reqCnr cid.ID, ownerCnr user.ID, reqSenderKey []byte, curEpoch uint64) error {
 	// 1. First check token lifetime. Simplest verification.
-	if !token.ValidAt(st.CurrentEpoch()) {
+	if !token.ValidAt(curEpoch) {
 		return errBearerExpired
 	}
 
 	// 2. Then check if bearer token is signed correctly.
-	_, err := icrypto.VerifyTokenSignature(token)
+	signer, err := icrypto.VerifyTokenSignature(&token)
 	if err != nil {
 		return fmt.Errorf("verify bearer token signature: %w", err)
 	}
+	if issuer := token.Issuer(); !issuer.IsZero() && signer != issuer {
```

Comment:
if `issuer.IsZero`, we continue? could not find info if missing issuer is ok there: https://github.com/nspcc-dev/neofs-api/commit/f49089fc59d0af2e2765c869e4438a44a47e29b6

Reply:
At this stage non-zero issuer can be enforced, I think. Bearer tokens are not saved anywhere long-term and I'd expect all of the old ones to expire already.

Reply:
initially was done for compatibility with tokens issued before https://github.com/nspcc-dev/neofs-api/commit/f49089fc59d0af2e2765c869e4438a44a47e29b6. Changed to the required issuer field